tests_end_to_end/tests/Datasets/datasets_utils.py

```diff
@@ -48,6 +48,7 @@ def insert_dataset_items_ui(page: Page, dataset_name, items_list):
     
     for item in items_list:
         dataset_items_page.insert_dataset_item(json.dumps(item))
+        time.sleep(0.2)
```

Comment:
Any particular reason to throttle the insertion pace here? The recommended alternative is inserting dataset items in batches, so you can insert the whole `items_list` in one go.

Reply:
The main bit of flakiness that's currently present in these tests is one dataset item you insert via the UI not registering if you switch away too quickly. Added a small delay to try and prevent that and see if it works. As far as I know there is no way to add a list of dataset items via the UI, only one at a time, which is why I do it this way here. When inserting via the SDK i just pass the entire list in one go